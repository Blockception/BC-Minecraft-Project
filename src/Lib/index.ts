/*	Auto generated	*/

export * from "./mcattributes";
export * from "./mcdefinitions";
export * from "./mcignore";
export * from "./project";