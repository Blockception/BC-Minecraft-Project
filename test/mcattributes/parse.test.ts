import { MCAttributes } from "../../src/main";
import { TestFilesFolder } from "../utillity.test";
import * as path from "path";
import assert = require("assert");

const Text1 = `diagnostics.objectives=true
diagnostics.tags=false
diagnostics.json=true`;

const Attribute1 = [
  { key: "diagnostics.objectives", value: "true" },
  { key: "diagnostics.tags", value: "false" },
  { key: "diagnostics.json", value: "true" },
];

describe("MCAttributes", () => {
  it("parse1", () => {
    let parse = MCAttributes.parse(Text1);

    for (var I = 0; I < Attribute1.length; I++) {
      const element = Attribute1[I];

      const value = parse[element.key];

      assert.strictEqual(value, element.value);
    }
  });

  it("loadSync file1", () => {
    const filepath = path.join(TestFilesFolder, "mcattributes", "file1.mcattributes");

    let Attributes = MCAttributes.loadSync(filepath);

    assert.strictEqual(Attributes["diagnose"], "true");
    assert.strictEqual(Attributes["diagnose.objectives"], "true");
    assert.strictEqual(Attributes["diagnose.tags"], "false");
    assert.strictEqual(Attributes["diagnose.mcfunctions"], "true");

    assert.strictEqual(Attributes["world.area_used"], "0 0 0 1000 256 1000");
  });

  it("load file2", (done) => {
    const filepath = path.join(TestFilesFolder, "mcattributes", "file2.mcattributes");

    MCAttributes.load(filepath).then((Attributes) => {
      assert.strictEqual(Attributes["diagnose"], "true");
      assert.strictEqual(Attributes["diagnose.objectives"], "true");
      assert.strictEqual(Attributes["diagnose.tags"], "false");
      assert.strictEqual(Attributes["diagnose.mcfunctions"], "true");

      assert.strictEqual(Attributes["world.area_used"], "0 0 0 1000 256 1000");

      done();
    });
  });
});
