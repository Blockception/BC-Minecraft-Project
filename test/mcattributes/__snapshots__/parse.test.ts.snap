// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
{
  "diagnostics.json": "true",
  "diagnostics.objectives": "true",
  "diagnostics.tags": "false",
}
`;

exports[`MCAttributes load file2 1`] = `
{
  "diagnose": "true",
  "diagnose.mcfunctions": "true",
  "diagnose.objectives": "true",
  "diagnose.tags": "false",
  "world.area_used": "0 0 0 1000 256 1000",
}
`;

exports[`MCAttributes loadSync file1 1`] = `
{
  "diagnose": "true",
  "diagnose.mcfunctions": "true",
  "diagnose.objectives": "true",
  "diagnose.tags": "false",
  "world.area_used": "0 0 0 1000 256 1000",
}
`;
